security/pkg/k8s/controller/workloadsecret.go

```diff
@@ -87,6 +91,12 @@ type SecretController struct {
 	// Length of the grace period for the certificate rotation.
 	gracePeriodRatio float32
 
+	// Whether controller loop should target namespaces without the NamespaceManagedLabel
+	manageUnlabeled bool
+
+	// Used to coordinate with label and check if this instance of Citadel should create secret
+	selfNamespace string
```

Comment:
I believe we received feedback on injector that we need to support finer granularity - please use either serviceName or URL to identify the CA that manages a namespace, and match against it. 

That would allow a canary CA in same namespace as the prod one.

Reply:
Reiterating the comment from @incfly [here](https://github.com/istio/istio/pull/15503#discussion_r303623713) wrt using a URL as the granularity of selection.

As for using `serviceName`, is there a good mechanism for a `Pod` to know what `Service` it's running under? Don't think it's as straightforward to expose this to the pod as it is to expose the `namespace` it's running under